src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -1080,16 +1081,21 @@ private void writeLogTo(WriteMethod method, OutputStream os) throws IOException
 
     @Override public File getLogFile() {
         LOGGER.log(Level.WARNING, "Avoid calling getLogFile on " + this, new UnsupportedOperationException());
-        try {
-            File f = File.createTempFile("deprecated", ".log", getRootDir());
-            f.deleteOnExit();
-            try (OutputStream os = new FileOutputStream(f)) {
-                getLogText().writeRawLogTo(0, os);
+        File f = super.getLogFile();
+        if (LogStorage.of(asFlowExecutionOwner()) instanceof FileLogStorage) {
+            // FileLogStorage does write a file with the same name and the same format as before JEP-210, so accept it if it is there.
+        } else {
+            // Some other storage. Cache under the traditional name, but load from the defined source on each call.
```

Comment:
Question for @jglick - can you please enumerate in code comments which of these cases are meant to handle:

1. Traditional log storage on-master
2. The new log-rewrite log storage on-master
3. Cloud storage on-master

And I feel like we may actually be making a mistake by handing responsibility for those to the `WorkflowJob` or indeed the `Job` - it feels like the LogStorage itself ought to take ownership for these (including tests for each case).

Reply:
> can you please enumerate in code comments which of these cases are meant to handle

I already did, but I can try to reword the comments.

> * Traditional log storage on-master
> * The new log-rewrite log storage on-master

There is no difference between these two so far as the `log` file is concerned.

> 3\. Cloud storage on-master

There is no such thing.

> the LogStorage itself ought to take ownership for these

No. `LogStorage` handles writes to, and read from, logs wherever they may be stored, and defines the API for making that pluggable. Legacy code which presumes that the log is a file on the master has to be accommodated somehow, for compatibility.

Reply:
> There is no such thing.

Heh, typing fast, but you know what I meant. :)

> No. LogStorage handles writes to, and read from, logs wherever they may be stored, and defines the API for making that pluggable

Per this: https://github.com/jenkinsci/jep/blob/master/jep/207/README.adoc#run-api-compatibility isn't the `LogStorage` method `File toLogFile()` supposed to be able to handle all of this? 

What, concretely, is the `else` case addressing here?  IIUC unless a key part was left un-implemented we might be able to remove most of this logic.

Reply:
And if there *is* a concrete case that we know the 'else' branch is covering, where is its testcase?

Reply:
JEP-207 is only implemented in an experimental branch and I did not propose that API.

The `else` case will be active when you run the `pipeline-log-fluentd-cloudwatch` plugin.